### PR TITLE
hooks

### PR DESCRIPTION
**Pull request by Arzamas-ka**
fix page `EB`

implement `No Movie Found`

setup `defaultImage` when `posterImg` from server is not found

load more movies (pages)

spinner

`MovieDetails` component

`MovieDetails` component responsive

### CODE REVIEW

File: src/components/Catalog/Posters/index.tsx [hunk 1]
```diff
@@ -1,33 +1,103 @@
...
+    target.src = defaultImgMovie;
+    target.alt = 'image not found';
+  }, []);
+
+  const posters = movies.movies.map((poster) => {
+    const genre = poster.genres.map((genre) => (
+      <span key={shortid.generate()}> {genre}, </span>
+    ));
+
+    return (
```
Comment (osonovaakylai) [inline src/components/Catalog/Posters/index.tsx:62]: Pls extract this part into separate Poster component

File: src/components/Catalog/Posters/index.tsx [hunk 2]
```diff
@@ -1,33 +1,103 @@
...
+const Posters: FC<PostersProps> = ({ setMovieDetails }: any) => {
+  const { movies, error, loading, fetchMovies }: any = usePostersFetch();
+  const {
+    movieDetails,
+    loadingMovieDetails,
+    errorMovieDetails,
+    fetchMovieDetails,
+  }: any = useMoreDetailsMovie();
+
+  const loadMorePosters = () => {
```
Comment (osonovaakylai) [inline src/components/Catalog/Posters/index.tsx:38]: UseCallback is needed in every inline function

File: src/ErrorBoundary/index.tsx [hunk 3]
```diff
@@ -1,9 +1,10 @@
-import React, { Component, ReactNode } from 'react';
-
-import { PropsEB, StateEB } from './models';
-
+import React, { Component } from 'react';
+// types and styles
```
[the remark this answers is not in the document]
Comment (Arzamas-ka) [reply src/ErrorBoundary/index.tsx]: removed

File: src/components/AddMoviePopup/index.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  movie: '',
+  genre: '',
+  overview: '',
+  runtime: '',
+};
+
+const AddMoviePopup: FC<AppMoviePopup> = ({ hide, isShowing }) => {
+  const [values, setValues] = useState(initialValues);
+
+  const handleOnChange = ({ target }) => {
```
[the remark this answers is not in the document]
Comment (Arzamas-ka) [reply src/components/AddMoviePopup/index.tsx]: wraped

File: src/components/AddMoviePopup/index.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    const value = target.type === 'checkbox' ? target.checked : target.value;
+
+    setValues({
+      ...values,
+      [target.name]: value,
+    });
+  };
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
```
[the remark this answers is not in the document]
Comment (Arzamas-ka) [reply src/components/AddMoviePopup/index.tsx]: wraped

File: src/components/AddMoviePopup/index.tsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+            type="text"
+            placeholder="Runtime here"
+            onChange={handleOnChange}
+            value={values.runtime}
+            autoComplete="off"
+          />
+        </StyledAddMoviePopupContainer>
+
+        <StyledButtonContainer>
+          <Button reset type="reset" onClick={null} text="Reset"></Button>
```
[the remark this answers is not in the document]
Comment (Arzamas-ka) [reply src/components/AddMoviePopup/index.tsx]: fixed

File: src/components/Catalog/Posters/models.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
+// export interface PostersProps {
```
[the remark this answers is not in the document]
Comment (Arzamas-ka) [reply src/components/Catalog/Posters/models.ts]: not need )

File: src/components/Select/index.tsx [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  StyledSelectTitle,
+  StyledSelectMain,
+} from './style';
+
+const Select: FC<SelectProps> = ({ onChange, name }) => {
+  return (
+    <StyledSelectWrapper>
+      <StyledSelectTitle>Genre</StyledSelectTitle>
+      <StyledSelectMain name={name} onChange={onChange}>
+        <option hidden>Select Genre</option>
```
[the remark this answers is not in the document]
Comment (Arzamas-ka) [reply src/components/Select/index.tsx:22]: moved

File: src/hooks/useMoreDetailsMovie.ts [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+      try {
+        const response = await axios.get(`${API_BASE}/${id}`);
+        const responseMovieDetails = response.data;
+
+        setMovieDetails(responseMovieDetails);
+      } catch (error) {
+        setErrorMovieDetails(true);
+      }
+
+      setLoadingMovieDetails(false);
```
[the remark this answers is not in the document]
Comment (Arzamas-ka) [reply src/hooks/useMoreDetailsMovie.ts]: created `finally` {}

File: src/hooks/usePopup.ts [hunk 10]
```diff
@@ -0,0 +1,13 @@
+import { useState } from 'react';
+
+const usePopup = () => {
+  const [isShowing, setIsShowing] = useState(false);
+
+  const toggle = () => {
```
[the remark this answers is not in the document]
Comment (Arzamas-ka) [reply src/hooks/usePopup.ts]: wraped

File: src/components/Catalog/Posters/index.tsx (hunk 1 above)
Comment (Arzamas-ka) [reply src/components/Catalog/Posters/index.tsx:34]: separated to PosterItem component

File: src/components/Catalog/Posters/index.tsx (hunk 2 above)
Comment (Arzamas-ka) [reply src/components/Catalog/Posters/index.tsx]: wraped

File: src/components/Select/index.tsx [hunk 11]
```diff
@@ -1,36 +1,29 @@
...
-      <SelectTitle>Genre</SelectTitle>
-      <SelectMain name="genre" onChange={handleChange}>
+    <StyledSelectWrapper>
+      <StyledSelectTitle>Genre</StyledSelectTitle>
+      <StyledSelectMain name={name} onChange={onChange}>
         <option hidden>Select Genre</option>
         {options}
-      </SelectMain>
-    </SelectStyled>
+        <option>Science Fiction</option>
```
Comment (osonovaakylai) [inline src/components/Select/index.tsx:24]: Seems like this option was forgotten

File: src/hooks/usePostersFetch.ts [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+      setMovies((prev) => ({
+        ...prev,
+        movies: [...prev.movies, ...posters.data],
+        currentPage: posters.offset,
+        totalPages: posters.totalAmount / posters.limit - posters.offset,
+      }));
+    } catch (error) {
+      setError(true);
+    }
+
```
Comment (osonovaakylai) [inline src/hooks/usePostersFetch.ts:31]: Finally block could be added here also

**Review by osonovaakylai — APPROVED** [review]

File: src/components/Select/index.tsx (hunk 11 above)
Comment (Arzamas-ka) [reply src/components/Select/index.tsx]: oops) fixed

File: src/hooks/usePostersFetch.ts (hunk 12 above)
Comment (Arzamas-ka) [reply src/hooks/usePostersFetch.ts]: done

**Pull request merged**